src/actions/clusterActions.js

```diff
@@ -290,45 +343,43 @@ export function clusterDeleteApp(appName, clusterID) {
  * @param {String} clusterId Cluster ID
  */
 export function clusterLoadDetails(clusterId) {
-  return function(dispatch, getState) {
+  return async function(dispatch, getState) {
     var token = getState().app.loggedInUser.auth.token;
     var scheme = getState().app.loggedInUser.auth.scheme;
+    const nodePoolsClusters = getState().entities.clusters.nodePoolsClusters;
+    const isNodePoolCluster = nodePoolsClusters.includes(clusterId);
 
     dispatch({
       type: types.CLUSTER_LOAD_DETAILS,
       clusterId,
     });
 
-    var cluster;
-    var clustersApi = new GiantSwarm.ClustersApi();
-
-    return clustersApi
-      .getCluster(scheme + ' ' + token, clusterId)
-      .then(c => {
-        cluster = c;
-        return dispatch(clusterLoadStatus(clusterId));
-      })
-      .then(() => {
-        cluster.capabilities = computeCapabilities(
-          cluster,
-          getState().app.info.general.provider
-        );
-        dispatch(clusterLoadDetailsSuccess(cluster));
-        return cluster;
-      })
-      .catch(error => {
-        console.error('Error loading cluster details:', error);
-        dispatch(clusterLoadDetailsError(clusterId, error));
-
-        new FlashMessage(
-          'Something went wrong while trying to load cluster details.',
-          messageType.ERROR,
-          messageTTL.LONG,
-          'Please try again later or contact support: support@giantswarm.io'
-        );
-
-        throw error;
-      });
+    try {
```

Comment:
Not a big fan of try/catch for larger chunks of code like this, since multiple things could potentially be the culprit here, but I am ok with seeing how it goes.